cmd/snapd-generator/main.c

```diff
@@ -97,3 +91,106 @@ int main(int argc, char **argv)
 	fprintf(f, "WantedBy=local-fs.target\n");
 	return 0;
 }
+
+static bool file_exists(const char *path) {
+	struct stat buf;
+	return stat(path, &buf) == 0;
```

Comment:
Please check that `path` is a file or symbolic link `(buf.st_mode & S_IFMT) == S_IFREG` and add a comment:
```
/* Not using lstat to automatically resolve symbolic links, including handling, as an error, dangling symbolic links. */
```

Reply:
Done.